### PR TITLE
[IMP] (website_)sale: add default salesperson to online sales orders 

### PR DESCRIPTION
**Pull request by ppr-odoo**
Pad:https://pad.odoo.com/p/r.b315258e687a63bdb1e6f4a2f0b6016b
Task:https://www.odoo.com/web#id=2059492&action=327&model=project.task&view_type=form&menu_id=4720

### CODE REVIEW

**Comment by AntoineVDV** [comment]
First diagonal pass done. LGTM

File: addons/website_sale/models/sale_order.py [hunk 1]
```diff
@@ -348,6 +348,17 @@ def _cart_recovery_email_send(self):
...
+    def _send_order_confirmation_mail(self):
+        affiliate_user = False
+        if request and request.session.get('affiliate_id'):
+            affiliate_id = request.session.get('affiliate_id')
+            affiliate_user = self.env['res.users'].sudo().browse(affiliate_id).exists()
+        for order in self.filtered(lambda l: l.state in ('sent', 'sale') and l.website_id):
+            salesperson_id = affiliate_user and affiliate_user.id or order.website_id.salesperson_id.id
+            user_id = order.partner_id.parent_id.user_id.id or order.partner_id.user_id.id
+            order.write({'user_id': salesperson_id or user_id or self.env.ref('base.user_admin').id})
+        return super()._send_order_confirmation_mail()
```
Comment (Feyensv) [inline addons/website_sale/models/sale_order.py]:
```python
affiliate_user = self.env['res.users']
if request and request.session.get('affiliate_id')
  affiliate_user = self.env['res.users'].sudo().browse(request.session.get('affiliate_id'))
for order in self.filtered(lambda l: ...)
  user = affiliate_user or order.website_id.salesperson_id or order.partner_id.user_id or order.partner_id.user_id
  order.user_id = user
```

Comment (ppr-odoo) [reply addons/website_sale/models/sale_order.py]: @Feyensv cc @mpa-odoo 
 I've changes done.

**Comment by ppr-odoo** [comment]
Task Cancelled 2323254

**Comment by AntoineVDV** [comment]
> Task Cancelled 2323254

Hello, are you sure that this is the right task id?

**Comment by ppr-odoo** [comment]
Task Cancelled 2059492 cc @AntoineVDV

**Pull request closed without merging**